### PR TITLE
CORE2 target added

### PR DESCRIPTION
**Pull request by byq77**
Hi,

Responding to: https://github.com/ARMmbed/mbed-os/pull/14898#issuecomment-876487668

CORE2 is a board used by [ROSbots](https://husarion.com/manuals/rosbot/). A firmware to these robots is open source and uses mbed-os 5.14.x, but I'll probably rewrite it soon to use the latest version of the lib. If you don't mind I can add a link to the firmware code later on. Might be useful for somebody who would want to develop a new application using this framework and looks for some use case in robotics.

### CODE REVIEW

**Comment by jeromecoutant** [comment]
> A firmware to these robots is open source and uses mbed-os 5.14.x, but I'll probably rewrite it soon to use the latest version of the lib. 

So this PR is supposed to address 5.15 maintenance branch,
or 6.x branch ?
(goal of this repo was supposed to address mbed-os-6 support)

**Comment by byq77** [comment]
This PR targets mbed 6.

File: TARGET_STM32F4/TARGET_CORE2/PinNames.h [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+    BUTTON2 = PG_13,
+
+    // I2C & SPI
+    I2C_SCL     = PB_8, /* I2C1 */
+    I2C_SDA     = PB_9,
+    SPI_MOSI    = PA_7,
+    SPI_MISO    = PA_6,
+    SPI_SCK     = PA_5,
+    SPI_CS      = PB_6,
+    PWM_OUT     = PB_3,
```
Comment (jeromecoutant) [inline TARGET_STM32F4/TARGET_CORE2/PinNames.h]: Pin aliases are supposed to be defined as #define xxx

see https://github.com/ARMmbed/mbed-os/blob/master/hal/include/hal/PinNameAliases.h#L160

Comment (byq77) [reply TARGET_STM32F4/TARGET_CORE2/PinNames.h]: Fixed in https://github.com/ARMmbed/stm32customtargets/pull/12/commits/dee4b5e3f980c85eefc5ee445f08d0913d990ac8.

**Comment by jeromecoutant** [comment]
I am sorry, I have updated the README file.
Please rebase on top of master (no merge commit)

**Comment by byq77** [comment]
Rebased. Should be ok now.

**Comment by jeromecoutant** [comment]
Oups, I merged #15 which makes a conflict again...

File: README.md [hunk 2]
```diff
@@ -173,6 +175,17 @@ Use RAK3172 and added
 - Green Led on PA10 (LED2) reversed (1=OFF, 0=ON)
 - SMD CR2450 battery holder
 
+# STM32F4
```
Comment (jeromecoutant) [inline README.md]: Minor comment :-)
Could you add STM32F4 after F1, before L0 ?

File: TARGET_STM32F4/CMakeLists.txt [hunk 3]
```diff
@@ -1,2 +1,4 @@
 # Copyright (c) 2020 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
+
+add_subdirectory(TARGET_CORE2 EXCLUDE_FROM_ALL)
```
Comment (jeromecoutant) [inline TARGET_STM32F4/CMakeLists.txt]: Maybe add a new line to make GitHub happy ?

**Comment by byq77** [comment]
Come to surface again ;-P

**Review by jeromecoutant — APPROVED** [review]

**Pull request merged**